core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -1450,7 +1457,95 @@ class GroupCoordinator(val brokerId: Int,
             group.maybeInvokeJoinCallback(member, joinResult)
             completeAndScheduleNextHeartbeatExpiration(group, member)
             member.isNew = false
+
+            group.addPendingSyncMember(member.memberId)
           }
+
+          schedulePendingSync(group)
+        }
+      }
+    }
+  }
+
+  private def removePendingSyncMember(
+    group: GroupMetadata,
+    memberId: String
+  ): Unit = {
+    group.removePendingSyncMember(memberId)
+    maybeCompleteSyncExpiration(group)
+  }
+
+  private def removeSyncExpiration(
```

Comment:
Definitely. I will add that.

I cannot really think of any other doable alternatives. At least, we know that the leader has sent the SyncGroup as the group transitioned to stable.

Throwing few ideas:
* Could we store the pending set in the log? That does not seem really doable as the set could be updated before the write is acknowledged. Therefore we could think that a member is pending after a failover but it might not be.
* We could transition to Stable only when all pending members have joined. We already discussed this and agreed that this is not ideal as it would delay processing.

Reply:
Yeah, I don't see a good option unless we want to add more state to the log, and I'm not really sure it's worth it. Maybe we can file a jira at least so we don't forget about the gap.

Reply:
JIRA: https://issues.apache.org/jira/browse/KAFKA-12962